src/charm.py

```diff
@@ -124,6 +124,24 @@ def is_main(self) -> bool:
         """
         return self.get_main_unit() == self.unit.name
 
+    def get_unit_number(self, unit_name: str = "") -> str:
+        """Get unit number from unit name.
+
+        Args:
+            unit_name: unit name. E.g.: synapse/0
+
+        Returns:
+            Unit number. E.g.: 0
+        """
+        if not unit_name:
+            unit_name = self.unit.name
+        match = re.search(r"[-\/](\d+)", unit_name)
+        # A Juju unit name is s always named on the
+        # pattern <application>/<unit ID>, where <application> is the name
```

Comment:
Would using regex a bit overkill if the pattern is this simple?

Reply:
Actually the comment is outdated. This pattern is for two strings:
synapse/0 and synapse-0.synapse.endpoints
Maybe my mind is too shell-oriented :laughing: 
How would you do without regex?

Reply:
Replaced by split + rfind
Please, if you want to comment something more regarding this I'll leave the https://github.com/canonical/synapse-operator/pull/266#discussion_r1540590003 open so you and Bartz can see it.